Create English locale

ivonasto opened the pull request:
Translation of the app in English, excluding data coming from the server.  I suggest we split the task in two. 

The app checks if the platform's language is supported, if not, it defaults to English. There is a button on the Settings page to change it manually. Its design is rudimentary, suggestions are welcome. 

Fixed a small mistake in the `app_rating_manager.dart` - the labels on the `no` and `maybe letter` buttons were interchanged.
![image](https://github.com/user-attachments/assets/481d418d-e5ff-4d5c-b1ea-7455d28afc2e)

I have doubts about the following translation: "същински трасета". I went with "main routes" could also be "official routes"? 

I tested it on Android only again.

ivonasto committed:
Create English locale

vakrilov commented on fivekmrun_app_flutter/lib/l10n/app_en.arb:
```diff
@@ -0,0 +1,138 @@
+    "login_preview_widget_loading": "loading ...",
+    "login_preview_widget_login_as": "Sign in as",
+
+    "best_time_by_route_chart_widget_records": "Records on main routes",
```
```suggestion
    "best_time_by_route_chart_widget_records": "Records on official routes",
```

etabakov replied:
Agree "official" seems more aligned with the spirit of "същински".

vakrilov commented on fivekmrun_app_flutter/lib/l10n/app_en.arb:
```diff
@@ -0,0 +1,138 @@
+    "app_rating_manager_no_button": "No, thank you",
+    "app_rating_manager_later_button": "Later",
+
+    "runs_by_route_chart_title": "Runs on main routes",
```
```suggestion
    "runs_by_route_chart_title": "Runs on official routes",
```

vakrilov commented:
Awesome job! 🎉  

Code looks good!

> I have doubts about the following translation: "същински трасета". I went with "main routes" could also be "official routes"?
I think "official run"/"official routes" is the better option.

I will take some time to test on IOS. 

Also, If there are any additional steps related to to generating translation files, can you please note them in the [README.md](https://github.com/5kmrun-bg/fivekmrun-app/blob/master/fivekmrun_app_flutter/README.md)

vakrilov commented on fivekmrun_app_flutter/lib/l10n/app_bg.arb:
```diff
@@ -0,0 +1,138 @@
+  "offline_chart_page_no_results":"Няма резултати",
+  "offline_chart_page_login":"Вход с парола",
+  "offline_chart_page_participation_in":"Участието в ",
+  "offline_chart_page_leaderboard_access":"класацията е достъпно вход с парола!",
```
```suggestion
  "offline_chart_page_leaderboard_access":" класацията е достъпно вход с парола!",
```

vakrilov commented on fivekmrun_app_flutter/lib/state/locale_provider.dart:
```diff
@@ -0,0 +1,44 @@
+    if (localeCode != null) {
+      _locale = Locale(localeCode);
+    } else {
+      Locale platformLocale = Locale(Platform.localeName);
```
While testing on IOS I noticed that when I set the language and region to Bulgaria the `Platform.localeName` is actually `bg_BG` and not `bg` (the one defined in `AppLocalizations.supportedLocales`). Check the screenshot.
So this piece of logic always selects `en`. Do you think there is way to only check the preferred language and not the region?

<img width="406" alt="Screenshot 2024-11-29 at 23 34 46" src="https://github.com/user-attachments/assets/952ce594-0402-4c79-8a25-f8c967d84bd6">

vakrilov commented on fivekmrun_app_flutter/lib/state/locale_provider.dart:
```diff
@@ -0,0 +1,44 @@
+      Locale platformLocale = Locale(Platform.localeName);
+      _locale = (AppLocalizations.supportedLocales.contains(platformLocale))
+          ? platformLocale
+          : Locale("en");
```
WDYT of leaving `bg` as the default language?
```suggestion
          : Locale("bg");
```

etabakov replied:
Yeah, it makes sense. We are targeting only Bulgarians and the app doesn't make sense for foreigners. Let's leave BG as the default.

ivonasto replied:
My thought process for having "en" as fallback was the following: if the device is neither in BG nor EN, chances are it belongs to somebody who does not speak Bulgarian. And having the translation makes the app available to foreigners living in Bulgaria.

vakrilov commented on fivekmrun_app_flutter/lib/state/locale_provider.dart:
```diff
@@ -0,0 +1,44 @@
+    if (localeCode != null) {
+      _locale = Locale(localeCode);
+    } else {
+      Locale platformLocale = Locale(Platform.localeName);
+      _locale = (AppLocalizations.supportedLocales.contains(platformLocale))
+          ? platformLocale
+          : Locale("en");
```
While testing on IOS I noticed that when I set the language and region to Bulgaria the `Platform.localeName` is actually `bg_BG` and not `bg` (the one defined in `AppLocalizations.supportedLocales`). Check the screenshot.
So this piece of logic always selects `en`. Do you think there is way to only check the preferred language and not the region?

<img width="406" alt="Screenshot 2024-11-29 at 23 34 46" src="https://github.com/user-attachments/assets/952ce594-0402-4c79-8a25-f8c967d84bd6">

ivonasto replied:
It should work now. If you don't like the string logic, please let me know and I'll try to come up with something else.

vakrilov reviewed (commented):
Tested on IOS and all looks good except the [default language select](https://github.com/5kmrun-bg/fivekmrun-app/pull/144/files#r1863949501)

etabakov commented:
Great job, @ivonasto and good catch with the rating dialog 🤦

ivonasto committed:
Update fivekmrun_app_flutter/lib/l10n/app_en.arb

change "main" to "official"

Co-authored-by: Alexander Vakrilov <alexander.vakrilov@gmail.com>

ivonasto committed:
Update fivekmrun_app_flutter/lib/l10n/app_en.arb

change "main" to "official"

Co-authored-by: Alexander Vakrilov <alexander.vakrilov@gmail.com>

ivonasto committed:
Update fivekmrun_app_flutter/lib/l10n/app_bg.arb

add indent

Co-authored-by: Alexander Vakrilov <alexander.vakrilov@gmail.com>

ivonasto committed:
Update fivekmrun_app_flutter/lib/l10n/app_en.arb

add indent

Co-authored-by: Alexander Vakrilov <alexander.vakrilov@gmail.com>

ivonasto committed:
Change default fallback language to BG

Co-authored-by: Alexander Vakrilov <alexander.vakrilov@gmail.com>

ivonasto committed:
Fix language selection 

- extract language code from platform locale string before checking for support

ivonasto committed:
Keep language preferences after logging out

etabakov committed:
Merge branch 'master' into translation
